src/main/kotlin/com/terraformation/backend/accelerator/ParticipantProjectSpeciesService.kt

```diff
@@ -109,12 +131,74 @@ class ParticipantProjectSpeciesService(
     }
   }
 
+  /**
+   * When a species deliverable status is updated to "approved", we save a snapshot of the species
+   * list and reference it in the deliverable submission
+   */
+  @EventListener
+  fun on(event: DeliverableStatusUpdatedEvent) {
+    if (event.newStatus !== SubmissionStatus.Approved) {
+      return
+    }
+
+    val deliverable = deliverablesDao.fetchOneById(event.deliverableId)
+    if (deliverable == null || deliverable.deliverableTypeId != DeliverableType.Species) {
+      return
+    }
+
+    val speciesData =
+        participantProjectSpeciesStore.fetchSpeciesForParticipantProject(event.projectId)
+    val rows: List<List<Pair<String, String>>> =
+        speciesData.map {
+          listOf(
+              "Project ID" to it.project.id.toString(),
+              "Species ID" to it.species.id.toString(),
+              "Status" to it.participantProjectSpecies.submissionStatus.jsonValue,
+              "Rationale" to (it.participantProjectSpecies.rationale ?: ""),
+              "Feedback" to (it.participantProjectSpecies.feedback ?: ""),
+              "Internal Comment" to (it.participantProjectSpecies.internalComment ?: ""),
+              "Native / Non-Native" to
+                  (it.participantProjectSpecies.speciesNativeCategory?.jsonValue ?: ""),
+              "Species Scientific Name" to it.species.scientificName,
+              "Species Common Name" to (it.species.commonName ?: ""),
+          )
+        }
+
+    val stream = ByteArrayOutputStream()
+    val streamWriter = OutputStreamWriter(stream)
+
+    CSVWriter(
+            streamWriter,
+            CSVWriter.DEFAULT_SEPARATOR,
+            CSVWriter.DEFAULT_QUOTE_CHARACTER,
+            CSVWriter.DEFAULT_ESCAPE_CHARACTER,
+            CSVWriter.RFC4180_LINE_END,
+        )
+        .use { csvWriter ->
+          csvWriter.writeNext(rows.first().map { it.first }.toTypedArray())
+          rows.forEach { row -> csvWriter.writeNext(row.map { it.second }.toTypedArray()) }
```

Comment:
There is an extension method `writeNext()` in `Csv.kt` that lets you pass a `List<Any?>` instead of an `Array<String?>`, which would remove the need for the `.toTypedArray()` here.